src/Pages/ManageCharactersPage/ManageCharactersPage.tsx

```diff
@@ -1,25 +1,23 @@
-import React, { useRef } from 'react';
-import '../Components/Display Attribute Cards/DisplayAttributeCards.styles.scss';
+import React, { useRef, useState } from 'react';
+import '../../Components/Display Attribute Cards/DisplayAttributeCards.styles.scss';
 import 'Pages/page.styles.scss';
 
-import Header from '../Components/Header/Header.component';
-import CopyToClipboard from '../Components/Toolbar/CopyToClipboard';
-import DeleteCharacter from '../Components/Toolbar/DeleteCharacter/DeleteCharacter';
-import DisplaySelected from '../Components/Selected/DisplaySelected';
-import { useState } from 'react';
-import { ToEditCharacter } from '../Components/Header/Navigation Links/navigationLinks';
-import DeleteAlert from '../Components/Toolbar/DeleteCharacter/DeleteAlert';
-import { usePushBackOnRefresh } from '../Components/CustomHooks/PushBackOnRefresh';
-import ComponentToPrint from 'Components/Display Attribute Cards/ComponentToPrint';
+import Header from '../../Components/Header/Header.component';
+import CopyToClipboard from '../../Components/Toolbar/CopyToClipboard';
+import DeleteCharacter from '../../Components/Toolbar/DeleteCharacter/DeleteCharacter';
+import DisplaySelected from '../../Components/Selected/Display Selected/DisplaySelected';
+import { ToEditCharacter } from '../../Components/Header/Navigation Links/navigationLinks';
+import DeleteAlert from '../../Components/Toolbar/DeleteCharacter/DeleteAlert';
+import { usePushBackOnRefresh } from '../../Components/CustomHooks/PushBackOnRefresh';
+import ComponentToPrint from 'Components/Display Attribute Cards/ComponentToPrint/ComponentToPrint';
 import ReactToPrint from 'react-to-print';
 import { Button } from 'react-bootstrap';
-import { useAppSelector } from 'Components/CustomHooks/reduxHooks';
+import { useAppSelector } from 'features/reduxHooks';
 
 const ManageCharactersPage = () => {
   const characterName = useAppSelector((state) => state.character.name);
   const divRef = useRef(null);
   const [showAlert, setShowAlert] = useState(false);
-
   usePushBackOnRefresh();
```

Comment:
```suggestion
  const character = useAppSelector((state) => state.character);
  const divRef = useRef(null);
  const [showAlert, setShowAlert] = useState(false);

  if (!character.id) {
    return <Navigate to={'/create-or-manage-page'} />;
  }
```
Something like this is one way you could remove the effect, there's a few different things you could do with at the navigation level with react-router though depending on what you need to do.

Reply:
Oh man, mind blown.
It renders a hidden redirect component, I had no idea.

I was considering putting it in a function with a name like navigateOnREfresh.
I worry that its purpose would be unclear to the casual glance but making it a function would be less clean.
Which do you think is better?

Reply:
Don't think abstracting it out is needed. Of it was a big deal across components I'd be looking at higher level abstractions. Like persisting the character id. Sorry I haven't got my head around the hole app but maybe moving some similar screens to the same route and just render the different screens based on if there's an character I'd